File: examples/network/device/example_f5xc_single/go/main.go
package main

import (
	"github.com/equinix/pulumi-equinix/sdk/go/equinix/networkedge"
	"github.com/pulumi/pulumi/sdk/v3/go/pulumi"
)

func main() {
	pulumi.Run(func(ctx *pulumi.Context) error {
		sv := networkedge.GetAccountOutput(ctx, networkedge.GetAccountOutputArgs{
			MetroCode: pulumi.String("SV"),
		}, nil)
		_, err := networkedge.NewDevice(ctx, "f5xc-single", &networkedge.DeviceArgs{
			Name:      pulumi.String("tf-f5xc"),
			ProjectId: pulumi.String("XXXXXX"),
			MetroCode: pulumi.String(sv.ApplyT(func(sv networkedge.GetAccountResult) (*string, error) {
				return &sv.MetroCode, nil
			}).(pulumi.StringPtrOutput)),
			TypeCode:     pulumi.String("F5XC"),
			SelfManaged:  pulumi.Bool(true),
			Byol:         pulumi.Bool(true),
			Connectivity: pulumi.String("INTERNET-ACCESS"),
			PackageCode:  pulumi.String("STD"),
			Notifications: pulumi.StringArray{
				pulumi.String("john@equinix.com"),
				pulumi.String("marry@equinix.com"),
				pulumi.String("fred@equinix.com"),
			},
			TermLength: pulumi.Int(1),
			AccountNumber: pulumi.String(sv.ApplyT(func(sv networkedge.GetAccountResult) (*string, error) {
				return &sv.Number, nil
			}).(pulumi.StringPtrOutput)),
			AclTemplateId:  pulumi.String("xxxx"),
			Version:        pulumi.String("9.2025.17"),
			InterfaceCount: pulumi.Int(8),
			CoreCount:      pulumi.Int(8),
			VendorConfiguration: pulumi.StringMap{
				"token":    pulumi.String("XXXXXXXXXX"),
				"hostname": pulumi.String("XXXX"),
			},
		})
		if err != nil {
			return err
		}
		return nil
	})
}

File: examples/network/device/example_f5xc_single/python/__main__.py
import pulumi
import pulumi_equinix as equinix

sv = equinix.networkedge.get_account_output(metro_code="SV")
f5_xc_single = equinix.networkedge.Device("f5xc-single",
    name="tf-f5xc",
    project_id="XXXXXX",
    metro_code=sv.metro_code,
    type_code="F5XC",
    self_managed=True,
    byol=True,
    connectivity="INTERNET-ACCESS",
    package_code="STD",
    notifications=[
        "john@equinix.com",
        "marry@equinix.com",
        "fred@equinix.com",
    ],
    term_length=1,
    account_number=sv.number,
    acl_template_id="xxxx",
    version="9.2025.17",
    interface_count=8,
    core_count=8,
    vendor_configuration={
        "token": "XXXXXXXXXX",
        "hostname": "XXXX",
    })
